src/main/java/com/tvd12/gamebox/octree/OcTreeNode.java

```diff
@@ -0,0 +1,205 @@
+package com.tvd12.gamebox.octree;
+
+import com.tvd12.gamebox.entity.PositionAware;
+import com.tvd12.gamebox.math.Bounds;
+import com.tvd12.gamebox.math.Vec3;
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class OcTreeNode<T extends PositionAware> {
+    
+    @Setter
+    private OcTreeNode<T> parentNode = null;
+    private final int maxItems;
+    private final float minNodeSize;
+    private final Bounds bounds;
+    private final Set<T> items = new HashSet<>();
+    private final List<OcTreeNode<T>> children = new ArrayList<>();
+    
+    private static final int NUM_CHILDREN = 8;
+
+    public OcTreeNode(Bounds bounds, int maxItems, float minNodeSize) {
+        this.bounds = bounds;
```

Comment:
```java
if (minNodeSize <= 0) {
            throw new IllegalArgumentException(
                "minNodeSize must > 0 to avoid StackOverflow"
            );
        }
```